src/main/java/org/wildfly/security/audit/PeriodicRotatingFileAuditEndpoint.java

```diff
@@ -0,0 +1,231 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.audit;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.TemporalAdjusters;
+import java.time.temporal.WeekFields;
+import java.util.Date;
+import java.util.Locale;
+
+import static org.wildfly.common.Assert.checkNotNullParam;
+import static org.wildfly.security._private.ElytronMessages.audit;
+
+/**
+ * An audit endpoint which rotates the log at the log at a preset time interval.
+ *
+ * Based on {@link org.jboss.logmanager.handlers.PeriodicSizeRotatingFileHandler}.
+ *
+ * @author <a href="mailto:jkalina@redhat.com">Jan Kalina</a>
+ * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
+ * @author <a href="mailto:yborgess@redhat.com">Yeray Borges</a>
+ */
+public class PeriodicRotatingFileAuditEndpoint extends FileAuditEndpoint {
+
+    private final DateTimeFormatter format;
+    private final Period period;
+    private final ZoneId timeZone;
+    private long nextRollover = Long.MAX_VALUE;
+    private String nextSuffix;
+
+    PeriodicRotatingFileAuditEndpoint(Builder builder) throws IOException {
+        super(builder);
+        this.format = builder.format;
+        this.period = builder.period;
+        this.timeZone = builder.timeZone;
+
+        final File file = getFile();
+        calcNextRollover(file != null && file.lastModified() > 0 ? file.lastModified() : System.currentTimeMillis());
+    }
+
+    @Override
+    protected void write(byte[] bytes) throws IOException {
+        super.write(bytes);
+    }
+
+    @Override
+    protected void preWrite(Date date) {
+        final long recordMillis = date.getTime();
+        if (recordMillis >= nextRollover) {
+            try {
+                final File file = getFile();
+                if (file == null) {
+                    // no file is set; a direct output stream or writer was specified
+                    return;
+                }
+                closeStreams(); // close the original file (some OSes won't let you move/rename a file that is open)
+                final Path target = Paths.get(file.getAbsolutePath() + nextSuffix);
```

Comment:
Is this safe on Windows?  Maybe it would be safer to instead say something like (from memory):
```
final Path target = file.toPath().resolve(file.getName() + nextSuffix);
```